tests/integration/test_gitignore_patterns.py

```
@@ -0,0 +1,286 @@
+"""
+Comprehensive unit tests for .gitignore patterns.
+
+Tests validate that the .gitignore file correctly excludes unwanted files
+while allowing necessary test artifacts and database files.
+"""
+
+import pytest
+import re
+from pathlib import Path
+from typing import List, Set
```

Ruff (0.14.6)

`typing.List` is deprecated, use `list` instead

(UP035)